### PR TITLE
feat: 66812 automatic task assignment

### PR DESCRIPTION
**Pull request by anna-ritense**
Documentation for PRs:
* https://github.com/valtimo-platform/valtimo-backend-libraries/pull/740
* https://github.com/valtimo-platform/valtimo-frontend-libraries/pull/490

### CODE REVIEW

File: using-valtimo/document/automatic-task-assignment.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# Automatic task assignment
+
+It is possible to have user tasks be automatically assigned to a Case Assignee making the tasks in a Case appear under 
+the Case Assignee's Task List.
+
+## Functionality
+
+This feature causes User Tasks (existing and new) to be assigned to the current case assignee if there is one assigned.  
+Upon the assignment or change of a user to a case all user tasks from all processes currently using the same business key will
```
Comment (ThomasRitense) [inline using-valtimo/document/automatic-task-assignment.md:9]: I think the first part of this sentence is not that easy to red. Maybe change to '(re-)assignment of a case'

File: using-valtimo/document/automatic-task-assignment.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
+# Automatic task assignment
+
+It is possible to have user tasks be automatically assigned to a Case Assignee making the tasks in a Case appear under
```
Comment (marijnritense) [inline using-valtimo/document/automatic-task-assignment.md:3]:
```suggestion
It is possible to have user tasks be automatically assigned to a case assignee making the tasks in a case appear under 
```

File: using-valtimo/document/automatic-task-assignment.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
+# Automatic task assignment
+
+It is possible to have user tasks be automatically assigned to a Case Assignee making the tasks in a Case appear under 
+the Case Assignee's Task List.
```
Comment (marijnritense) [inline using-valtimo/document/automatic-task-assignment.md:4]:
```suggestion
the case assignee's task list.
```

File: using-valtimo/document/automatic-task-assignment.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+## Functionality
+
+This feature causes User Tasks (existing and new) to be assigned to the current case assignee if there is one assigned.
```
Comment (marijnritense) [inline using-valtimo/document/automatic-task-assignment.md:8]:
```suggestion
This feature causes user tasks (existing and new) to be assigned to the current case assignee if there is one assigned.  
```

File: using-valtimo/document/automatic-task-assignment.md [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+using the same business key.
+
+## Prerequisites
+
+This feature requires the following:
+- Valtimo 10.7.0 or newer
+
+## How to configure automatic assignment of tasks
+
+Automatic assignment of tasks can be enabled at the bottom of a Case Configuration page (`Admin` -> `Case` ->
```
Comment (marijnritense) [inline using-valtimo/document/automatic-task-assignment.md:21]:
```suggestion
Automatic assignment of tasks can be enabled at the bottom of an individual case configuration page (`Admin` -> `Case` -> 
```

File: using-valtimo/document/automatic-task-assignment.md [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+## How to configure automatic assignment of tasks
+
+Automatic assignment of tasks can be enabled at the bottom of a Case Configuration page (`Admin` -> `Case` -> 
+`[your case definition]`). This option becomes visible when the Case is allowed to have an assignee.
+
+![Configuring automatic task assignment](img/case-automatic-task-assignment.png)
+
+## Limitations
+
+* Only tasks that the user has access to (User has the required role) are assigned to the case assignee
```
Comment (marijnritense) [inline using-valtimo/document/automatic-task-assignment.md:28]:
```suggestion
* Only tasks that the user has access to (user has the required role) are assigned to the case assignee
```

File: release-notes/major10/10.7.0/valtimo-backend-libraries.md [hunk 7]
```diff
@@ -4,10 +4,10 @@
 
 The following features were added:
 
-* **New feature1**
+* **Automatic User Task assignment to Case Handler**
 
-  Description of the new feature goes here.
-  Also link to the page that explains the feature in greater detail.
+  It is now possible to have User Tasks automatically assigned to a Case Handler
```
Comment (marijnritense) [inline release-notes/major10/10.7.0/valtimo-backend-libraries.md:9]:
```suggestion
  It is now possible to have user tasks automatically assigned to a case handler.
```

File: release-notes/major10/10.7.0/valtimo-frontend-libraries.md [hunk 8]
```diff
@@ -4,9 +4,16 @@
...
 
-* **Added edit note and delete note options on the notes tab**
+**Added edit note and delete note options on the notes tab**
 
-  Added the functionality to edit and delete a note in an order. It can only be edited if the text field is not empty.
+Added the functionality to edit and delete a note in an order. It can only be edited if the text field is not empty.
+
+* **Automatic User Task assignment to Case Handler**
+
+  An additional checkbox has been added to the Case Assignee section of a Case Configuration that enables the user to
```
Comment (marijnritense) [inline release-notes/major10/10.7.0/valtimo-frontend-libraries.md:13]:
```suggestion
  An additional checkbox has been added to the case assignee section of a case configuration that enables the user to 
```

File: release-notes/major10/10.7.0/valtimo-frontend-libraries.md [hunk 9]
```diff
@@ -4,9 +4,16 @@
...
-* **Added edit note and delete note options on the notes tab**
+**Added edit note and delete note options on the notes tab**
 
-  Added the functionality to edit and delete a note in an order. It can only be edited if the text field is not empty.
+Added the functionality to edit and delete a note in an order. It can only be edited if the text field is not empty.
+
+* **Automatic User Task assignment to Case Handler**
+
+  An additional checkbox has been added to the Case Assignee section of a Case Configuration that enables the user to 
+  toggle on and off the automatic assignment of User Tasks to a Case Handler if a Case has one assigned to it/is
```
Comment (marijnritense) [inline release-notes/major10/10.7.0/valtimo-frontend-libraries.md:14]:
```suggestion
  toggle on and off the automatic assignment of user tasks to a case handler if a case has one assigned to it/is 
```

File: using-valtimo/document/automatic-task-assignment.md
Comment (marijnritense) [inline using-valtimo/document/automatic-task-assignment.md:1]: Please also add this file to `SUMMARY.md`.

File: using-valtimo/plugin/catalogi-api/configure-catalogi-api-plugin.md [hunk 10]
```diff
@@ -64,3 +64,46 @@ When creating a process link the following properties have to be entered:
...
+The Catalogi API plugin supports the following actions that can be configured in process links in order to retrieve type 
+data.
+
+A general description on how to create process links can be found [here](../create-process-link.md).
+
+### Retrieve statustype
+This action retrieves the statustype URL using the statustype description, and will assign it to a process variable. 
+
+When creating a process link the following properties have to be entered:
+- **Statustype**. This description will be used to lookup the full statustype URL
```
Comment (marijnritense) [inline using-valtimo/plugin/catalogi-api/configure-catalogi-api-plugin.md:79]:
```suggestion
- **Statustype**. This description will be used to lookup the full statustype URL.
```

File: using-valtimo/plugin/catalogi-api/configure-catalogi-api-plugin.md [hunk 11]
```diff
@@ -64,3 +64,46 @@ When creating a process link the following properties have to be entered:
...
+data.
+
+A general description on how to create process links can be found [here](../create-process-link.md).
+
+### Retrieve statustype
+This action retrieves the statustype URL using the statustype description, and will assign it to a process variable. 
+
+When creating a process link the following properties have to be entered:
+- **Statustype**. This description will be used to lookup the full statustype URL 
+
```
Comment (marijnritense) [inline using-valtimo/plugin/catalogi-api/configure-catalogi-api-plugin.md:80]: I think this empty line can be removed

File: using-valtimo/plugin/catalogi-api/configure-catalogi-api-plugin.md [hunk 12]
```diff
@@ -64,3 +64,46 @@ When creating a process link the following properties have to be entered:
...
+  A full URL can also be set here. In that case, the value will be used directly without a lookup.
+- **Process variable name**. This is the name of the process variable which value will be set to the found resultaattype URL.
+
+An example action configuration:
+![Retrieve statustype](img/retrieve-resultaattype.png)
+### Retrieve besluittype
+This action retrieves the besluittype URL using the besluittype description, and will assign it to a process variable.
+
+When creating a process link the following properties have to be entered:
+- **Besluittype**. This description will be used to lookup the full besluittype URL
```
Comment (marijnritense) [inline using-valtimo/plugin/catalogi-api/configure-catalogi-api-plugin.md:102]:
```suggestion
- **Besluittype**. This description will be used to lookup the full besluittype URL.
```

File: using-valtimo/plugin/catalogi-api/configure-catalogi-api-plugin.md [hunk 13]
```diff
@@ -64,3 +64,46 @@ When creating a process link the following properties have to be entered:
...
+
+An example action configuration:
+![Retrieve statustype](img/retrieve-statustype.png)
+
+### Retrieve resultaattype
+This action retrieves the resultaattype URL using the resultaattype description, and will assign it to a process variable.
+
+When creating a process link the following properties have to be entered:
+- **Resultaattype**. This description will be used to lookup the full resultaattype URL
+
```
Comment (marijnritense) [inline using-valtimo/plugin/catalogi-api/configure-catalogi-api-plugin.md:92]: Newline can be removed

File: using-valtimo/plugin/catalogi-api/configure-catalogi-api-plugin.md [hunk 14]
```diff
@@ -64,3 +64,46 @@ When creating a process link the following properties have to be entered:
...
+- **Process variable name**. This is the name of the process variable which value will be set to the found statustype URL.
+
+An example action configuration:
+![Retrieve statustype](img/retrieve-statustype.png)
+
+### Retrieve resultaattype
+This action retrieves the resultaattype URL using the resultaattype description, and will assign it to a process variable.
+
+When creating a process link the following properties have to be entered:
+- **Resultaattype**. This description will be used to lookup the full resultaattype URL
```
Comment (marijnritense) [inline using-valtimo/plugin/catalogi-api/configure-catalogi-api-plugin.md:91]:
```suggestion
- **Resultaattype**. This description will be used to lookup the full resultaattype URL.
```

**Review by marijnritense — APPROVED** [review]

**Pull request merged**